pkg/dependency/parser/java/pom/parse.go

```diff
@@ -362,9 +371,11 @@ func (p *Parser) analyze(ctx context.Context, pom *pom, opts analysisOptions) (a
 		opts.exclusions = set.New[string]()
 	}
 	// Update remoteRepositories
-	pomReleaseRemoteRepos, pomSnapshotRemoteRepos := pom.repositories(p.servers)
-	p.releaseRemoteRepos = lo.Uniq(append(pomReleaseRemoteRepos, p.releaseRemoteRepos...))
-	p.snapshotRemoteRepos = lo.Uniq(append(pomSnapshotRemoteRepos, p.snapshotRemoteRepos...))
+	pomRepos := pom.repositories(p.servers)
+	p.remoteRepos.pom = lo.UniqBy(append(pomRepos, p.remoteRepos.pom...), func(r repository) string {
+		return r.url
+
```

Comment:
```suggestion
		return r.url
```

Reply:
used url.URL for `repository` https://github.com/aquasecurity/trivy/pull/9708/commits/6edca24d8bf8ad043d2e8958dd16d22e90b00a1b